src/nvim/os/shell.c

```diff
@@ -70,6 +73,25 @@ char **shell_build_argv(const char *cmd, const char *extra_args)
   return rv;
 }
 
+/// Builds the argument vector for any command with with arguments.
+///
+/// Creates: ["command", "that", "has", "spaces"] from a command string
+///
+/// @param cmd Command string
+/// @return A newly allocated argument vector. It must also be freed with
+///         `shell_free_argv` when no longer needed.
+char **build_argv(char *cmd)
+  FUNC_ATTR_NONNULL_RET FUNC_ATTR_MALLOC
+{
+  size_t argc = tokenize((char_u *)cmd, NULL);
+  char **rv = xmalloc((argc + 4) * sizeof(*rv));
+
+  tokenize((char_u *)cmd, rv);
+  assert(rv[0]);
```

Comment:
The value of `rv[0]` is random at this point. You could use `xcalloc((argc + 4) * sizeof(*rv),1)` instead of `xmalloc()` here to set the memory to `0` and trigger the `assert()`. 

 I would remove the `assert()`, because the code never writes a `NULL` into `rv` up to this point.